tools/c7n_mailer/c7n_mailer/sqs_message_processor.py

```diff
@@ -40,37 +39,33 @@ class SqsMessageProcessor(object):
      - set of resources from push policy, send to sns/explicit address
     """
 
-    def __init__(self, config, session, cache, logger):
+    def __init__(self, config, session, logger):
         self.config    = config
         self.logger    = logger
         self.aws_ses   = session.client('ses', region_name=config.get('ses_region'))
         self.aws_sts   = session.client('sts')
         self.sns_cache = {}
-        self.cache     = cache
         self.env       = jinja2.Environment(
             trim_blocks=True, autoescape=False)
         self.env.filters['yaml_safe'] = yaml.safe_dump
         self.env.filters['date_time_format'] = date_time_format
+        self.env.filters['get_date_time_delta'] = get_date_time_delta
 
         self.env.globals['format_resource'] = resource_format
         self.env.globals['format_struct'] = format_struct
         self.env.globals['resource_tag'] = resource_tag
         self.env.globals['resource_owner'] = resource_owner
 
-        self.OwnerTags = config['contact_tags']
-        self.get_user = partial(
-            get_user,
-            ldap_uri=config['ldap_uri'],
-            base_dn=config['ldap_bind_dn'],
-            ldap_bind_user=config['ldap_bind_user'],
-            ldap_bind_password=config['ldap_bind_password'])
+        self.OwnerTags = config.get('contact_tags', [])
+        self.ldap_username_tags = config.get('ldap_username_tags', [])
+        self.ldap_lookup = LdapLookup(config, logger)
 
-        self.env.loader = jinja2.FileSystemLoader(
+        self.env.loader = jinja2.FileSystemLoader([
             os.path.abspath(
                 os.path.join(
                     os.path.dirname(os.path.abspath(__file__)),
                     '..',
-                    'msg-templates')))
+                    'msg-templates')), os.path.abspath('/')])
```

Comment:
allow someone to specify the absolute directory of their jinja template file. This way you're not locked into the msg-templates directory in c7n_mailer. You can just set 'template: /tmp/foo.jinja' as an example.